fhir/com.b2international.snowowl.fhir.core/src/com/b2international/snowowl/fhir/core/request/codesystem/FhirRequest.java

```diff
@@ -120,12 +125,78 @@ protected String configureSummary() {
 		return Summary.TRUE;
 	}
 
-	public static final String extractLocales(CodeType displayLanguage) {
-		String locales = displayLanguage != null ? displayLanguage.getCode() : null;
-		if (CompareUtils.isEmpty(locales)) {
-			locales = AcceptLanguageHeader.DEFAULT_ACCEPT_LANGUAGE_HEADER;
+	/**
+	 * Converts a BCP-47 language tag (where the private use extension portions
+	 * are split into at most 8 characters, separated by dashes) into the "compact"
+	 * non-standard representation used in Snow Owl's internal API.
+	 * 
+	 * @param localeAsCode
+	 * @return
+	 */
+	public static final String compactLocale(final CodeType localeAsCode) {
```

Comment:
For these we need unit tests where the input should cover the BCP-47 and non-BCP-47 cases and similarly for the other helper functions.

Reply:
Added in https://github.com/b2ihealthcare/snow-owl/pull/1339/commits/bf40e8449506fb71876e17c196a7048b5e2aadbc.